src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Startup.cs

```diff
@@ -98,12 +99,12 @@ public void ConfigureServices(IServiceCollection services)
         // cause the data source builder to throw a host exception.
         if (!hostEnvironment.IsIntegrationTest())
         {
-            var connectionString = configuration.GetConnectionString("PublicDataDb")!;
+            var connectionString = ConnectionUtils.GetPostgreSqlConnectionString("PublicDataDb")!;
```

Comment:
Just wondering if you've tested a combination of this where you make the pipeline change to set the connection string using `az webapp config connection-string` but don't make this change to the Admin/Public API or Processor functions, to see if this change is required?

Based on the table shown in section [Connection string prefixes](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-8.0#connection-string-prefixes) , if there's an environment variable `CUSTOMCONNSTR_{KEY}` there should also be a converted configuration key created by removing the environment variable prefix and adding a configuration key section `ConnectionStrings:{KEY}`. If that's the case then I think this might have worked before, and our `ConnectionUtils` method of accessing the environment variables with the prefix could be redundant?

Reply:
I think what this is saying is that you would provide the connection string as a standard appsetting rather than explicitly as a connection string setting, but prefix it manually with one of the magic prefixes.  So it would be like:

```
                az functionapp config appsettings set \
                  --name '$(dataProcessorFunctionAppName)' \
                  --resource-group '$(resourceGroupName)' \
                  --slot 'staging' \
                  --settings \
                    "CoreStorage=@Microsoft.KeyVault(VaultName=$(keyVaultName); SecretName=$(coreStorageConnectionStringSecretKey))" \
                    "PublicDataDb=POSTGRESQLCONNSTR_@Microsoft.KeyVault(VaultName=$(keyVaultName); SecretName=$(dataProcessorPsqlConnectionStringSecretKey))"
                  
```